pdns/auth-carbon.cc

```diff
@@ -15,52 +15,54 @@ try
 {
   extern StatBag S;
 
-  for(int numloops=0;;++numloops) {
-    if(arg()["carbon-server"].empty()) {
+  string hostname=arg()["carbon-ourname"];
+  if(hostname.empty()) {
+    char tmp[80];
+    memset(tmp, 0, sizeof(tmp));
+    gethostname(tmp, sizeof(tmp));
+    char *p = strchr(tmp, '.');
+    if(p) *p=0;
+    hostname=tmp;
+    boost::replace_all(hostname, ".", "_");
+  }
+
+  vector<string> carbonServers;
+  stringtok(carbonServers, arg()["carbon-server"], ", ");
+
+  for(;;) {
+    if(carbonServers.empty()) {
       sleep(1);
       continue;
     }
-    if(numloops)
-      sleep(arg().asNum("carbon-interval"));
 
-    try {
-      ComboAddress remote(arg()["carbon-server"], 2003);
+    string msg;
+    msg.clear();
+
+    for (const auto& carbonServer : carbonServers) {
+      ComboAddress remote(carbonServer, 2003);
       Socket s(remote.sin4.sin_family, SOCK_STREAM);
-      
       s.setNonBlocking();
       s.connect(remote);  // we do the connect so the attempt happens while we gather stats
-      
+
       vector<string> entries = S.getEntries();
-      
-      ostringstream str;
-      time_t now=time(0);
-      string hostname=arg()["carbon-ourname"];
-      if(hostname.empty()) {
-	char tmp[80];
-	memset(tmp, 0, sizeof(tmp));
-	gethostname(tmp, sizeof(tmp));
-	char *p = strchr(tmp, '.');
-	if(p) *p=0;
-	hostname=tmp;
-	boost::replace_all(hostname, ".", "_");
-      }
-      for(const string& entry :  entries) {
-	str<<"pdns."<<hostname<<".auth."<<entry<<' '<<S.read(entry)<<' '<<now<<"\r\n";
+
+      if (msg.empty()) {
+        ostringstream str;
```

Comment:
This could be done before entering the `for (const auto& carbonServer : carbonServers)` loop, it might perhaps be easier to read.